Add Wagtail installation instructions

amakarudze opened the pull request:
Changes in this PR
- Add Wagtail to Tutorial Extensions

thibaudcolas commented on en/add_wagtail_to_your_website/wagtail_integration_adding_posts/README.md:
```diff
@@ -0,0 +1,130 @@
+model `Page` already contains a field by that name and hence this will cause an error. 
+- Our `Post` model also has fields`author` and `published_date` which we omitted in our `BlogPage` model because our 
+base model `Page` has a field `owner`that represents the `author` and has a `-first_published_at` field that represents 
+the `published_date`. 
```
Not sure we want to mention this but it’s pretty common for Wagtail page models to also have an author field. `owner` is useful but it’s not always appropriate as it can’t be set in the CMS. Adding an `author` field tends to be a good thing to cover in a Wagtail site tutorial, as it’s usually done with a vanilla Django model disconnected from `User` – always interesting to see how to manage those vanilla models in Wagtail compared to Django.

amakarudze replied:
Noted.

thibaudcolas commented on en/add_wagtail_to_your_website/wagtail_integration_adding_posts/README.md:
```diff
@@ -0,0 +1,130 @@
+As you can see, Wagtail takes care of most of the work we would have done for ourselves by writing our own code to 
+have that functionality. Among Wagtail's many powerful capabilities is the search capability for which Wagtail provides 
+two backends for storing the search index and performing search queries: one using the database's full-text search 
+capabilities, and another using Elasticsearch. You can also roll your own search backend. Isn't that awesome?
```
😄

thibaudcolas commented on en/add_wagtail_to_your_website/wagtail_integration_adding_posts/README.md:
```diff
@@ -0,0 +1,130 @@
+
+Wagtail’s built-in `get_parent()` method  is used to obtain the URL of the blog this post is a part of.
+
+Now we can create a few blog posts as children of `Home Page`. To do this, navigate to `Pages` and click on the arrow
```
Wagtail has pretty good documentation of the CMS interface if you wanted to refer people to it: https://guide.wagtail.org/en-latest/

Though it’s not available in that many languages currently, compared to Django Girls tutorials.

amakarudze replied:
Will add it.

thibaudcolas commented on en/add_wagtail_to_your_website/change_website_homepage/README.md:
```diff
@@ -0,0 +1,55 @@
+
+![Old Blog's New URL](images/old_blog_url.png)
+
+That's all for this tutorial. If you want to learn more about Wagtail, you can read the 
```
All looking excellent :)

thibaudcolas reviewed (changes requested):
Love it @amakarudze 😸 I think you did a great job at producing something where people can see the potential of Wagtail without it being super long.

Aside from my comments, I’d suggest to:

- Make sure there’s no further changes needed to push this to PythonAnywhere? I’m surprised it’d all "just work" there, but maybe it’s me.
- Update how the blog posts are fetched to only show live blog pages, and show them ordered by date.

And as quick wins that aren’t really needed but would be cool to demo:

- Adding the Wagtail userbar. Nice little feature.
- Perhaps making use of `parent_page_types` or `subpage_types`. This helps people understand the "child" pages quite a bit.

As a follow-up after this is merged, I’d love to know what you think of adding even more sections.

thibaudcolas reviewed (changes requested):
Love it @amakarudze 😸 I think you did a great job at producing something where people can see the potential of Wagtail without it being super long.

Aside from my comments, I’d suggest to:

- Make sure there’s no further changes needed to push this to PythonAnywhere? I’m surprised it’d all "just work" there, but maybe it’s me.
- Update how the blog posts are fetched to only show live blog pages, and show them ordered by date.

And as quick wins that aren’t really needed but would be cool to demo:

- Adding the Wagtail userbar. Nice little feature.
- Perhaps making use of `parent_page_types` or `subpage_types`. This helps people understand the "child" pages quite a bit.

As a follow-up after this is merged, I’d love to know what you think of adding even more sections.

amakarudze commented on en/add_wagtail_to_your_website/install_wagtail/README.md:
```diff
@@ -0,0 +1,179 @@
+# Wagtail Installation and Configuration
+
+> Part of this chapter is based on the WagTail tutorial by 
+> TorchBox (https://docs.wagtail.org/en/stable/getting_started/integrating_into_django.html).
```
Will remove this.

amakarudze commented on en/add_wagtail_to_your_website/wagtail_integration_adding_posts/README.md:
````diff
@@ -0,0 +1,130 @@
+```
+
+Before we move on, we will make some comparisons between our `Post` and `BlogPage` models. 
+- Our `Post` model contains a field named but we can't use the same field name in our `BlogPage` model because our base 
+model `Page` already contains a field by that name and hence this will cause an error. 
````
Oh yeah, it's `title` but it was omitted my mistake. Will add it.

amakarudze commented:
> Love it @amakarudze 😸 I think you did a great job at producing something where people can see the potential of Wagtail without it being super long.
> 
> Aside from my comments, I’d suggest to:
> 
> * Make sure there’s no further changes needed to push this to PythonAnywhere? I’m surprised it’d all "just work" there, but maybe it’s me.
> * Update how the blog posts are fetched to only show live blog pages, and show them ordered by date.
> 
> And as quick wins that aren’t really needed but would be cool to demo:
> 
> * Adding the Wagtail userbar. Nice little feature.
> * Perhaps making use of `parent_page_types` or `subpage_types`. This helps people understand the "child" pages quite a bit.
> 
> As a follow-up after this is merged, I’d love to know what you think of adding even more sections.

@thibaudcolas I addressed most of these points except for the quick wins that aren't really needed. They could be added later on as follow up.

thibaudcolas commented:
🤘I’ll do my best to review this soon

thibaudcolas commented:
Tagging @shyusu4 who has expressed an interest in reviewing this as part of her learning experience with Django & Wagtail 😌

shyusu4 reviewed (commented):
Great work and thank you for this! I think it’ll be really helpful for folks to get a headstart with how to use Wagtail CMS :)
I added a few comments and have an extra suggestion: I think it would be helpful if you added a reminder at the beginning to run `python manage.py runserver`, run migrations and create a superuser for Wagtail admin login so that people know how to get the project running - or so that it's clear that to run the project you need to follow some of the same steps as in the main tutorial.

thibaudcolas commented:
Thank you for giving this a go @shyusu4 😌 I’m still planning to review this myself too (but have to find the time somehow…)

cc my colleague @Morsey187 who is also interested in helping with reviewing this 🚀

thibaudcolas commented on en/add_wagtail_to_your_website/install_wagtail/README.md:
````diff
@@ -0,0 +1,179 @@
+To install Wagtail, type the following command in the command prompt:
+
+```
+(myvenv) ~/djangogirls$ pip install wagtail
````
Can confirm this will break at some point, as Wagtail 6.0 due for release in February 2024 will drop Django 3.2 compatibility. So for now I’d recommend:

```suggestion
(myvenv) ~/djangogirls$ pip install wagtail>=5.2,<5.3
```

amakarudze commented on en/add_wagtail_to_your_website/wagtail_integration_adding_posts/README.md:
```diff
@@ -0,0 +1,128 @@
+
+In the last section, we created a new page model which we used to add content to our homepage called `Homepage`, changed 
+the root of our Wagtail site to use this new homepage. We also added the CSS styling we were already using in our 
+project and now the Wagtail has the same look and feel as our website, which is really pretty amazing!
```
Yeah, I left `homepage`. Thanks for that.
